mllib/src/main/scala/org/apache/spark/ml/Evaluator.scala

```diff
@@ -36,5 +35,18 @@ abstract class Evaluator extends Identifiable {
    * @param paramMap parameter map that specifies the input columns and output metrics
    * @return metric
    */
-  def evaluate(dataset: DataFrame, paramMap: ParamMap): Double
+  def evaluate(dataset: DataFrame, paramMap: ParamMap): Double = {
+    this.copy(paramMap).evaluate(dataset)
+  }
+
```

Comment:
One question. Why `Evaluator` returns type `Double`? For example, confusion matrix can be an evaluator as well, and it can not be represented by type `Double`. Also, in our use-case, we often evaluate the models by the histogram of recommended popular titles, so it will be `Array[Long]`. Why don't we have type `T` here, and we specify the type in the implementation?